cmake/modules/SetUpFreeBSD.cmake

```diff
@@ -0,0 +1,149 @@
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+set(ROOT_PLATFORM freebsd)
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES x86_64 OR CMAKE_SYSTEM_PROCESSOR MATCHES amd64)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL Intel)
+    set(ROOT_ARCHITECTURE freebsdx8664icc)
+  else()
+    set(ROOT_ARCHITECTURE freebsdx8664)
+  endif()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES i686)
+  set(FP_MATH_FLAGS "-msse2 -mfpmath=sse")
+  if(CMAKE_CXX_COMPILER_ID STREQUAL Intel)
+    set(ROOT_ARCHITECTURE freebsdicc)
+  else()
+    set(ROOT_ARCHITECTURE freebsd)
+  endif()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES aarch64)
+  set(ROOT_ARCHITECTURE freebsdarm64)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES arm)
+  set(ROOT_ARCHITECTURE freebsdarm)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES ppc64 OR
+       CMAKE_SYSTEM_PROCESSOR MATCHES powerpc64 OR
+       CMAKE_SYSTEM_PROCESSOR MATCHES powerpc64le)
+  set(ROOT_ARCHITECTURE freebsdppc64gcc)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES s390x)
+  set(ROOT_ARCHITECTURE freebsds390xgcc)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES s390)
+  set(ROOT_ARCHITECTURE freebsds390gcc)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES riscv64)
+  set(ROOT_ARCHITECTURE freebsdriscv64)
+else()
+  message(FATAL_ERROR "Unknown processor: ${CMAKE_SYSTEM_PROCESSOR}")
+endif()
+
+# JIT must be able to resolve symbols from all ROOT binaries.
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -rdynamic")
+
+# Set developer flags
+if(dev)
+  # Warnings are errors.
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+
+  # Do not relink just because a dependent .so has changed.
+  # I.e. relink only if a header included by the libs .o-s has changed,
+  # whether or not that header "belongs" to a different .so.
+  set(CMAKE_LINK_DEPENDS_NO_SHARED On)
+
+  # Split debug info for faster builds.
+  if(NOT gnuinstall)
+    # We won't install DWARF files.
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -gsplit-dwarf")
+  endif()
+
+  # Try faster lld:
+  execute_process(COMMAND ${CMAKE_C_COMPILER} -fuse-ld=lld -Wl,--version OUTPUT_VARIABLE stdout ERROR_QUIET)
+  if("${stdout}" MATCHES "LLD ")
+    set(SUPERIOR_LINKER "lld")
+  endif()
```

Comment:
`lld` is the default on FreeBSD anyway, no?

Reply:
indeed, cleaning up